packages/gasket-plugin-metadata/lib/actions.js

```diff
@@ -1,13 +1,16 @@
-const path = require('path');
+import path from 'path';
+import { createRequire } from 'module';
+const require = createRequire(import.meta.url);
+
 const isModulePath = /^[/.]|^[a-zA-Z]:\\|node_modules/;
 const isGasketModule = /(@gasket\/|gasket-)(?!plugin)(?!preset).+/;
 const isGasketPreset = /(gasket-preset)|(@gasket\/preset-)/;
 const isGasketPlugin = /(gasket-plugin)|(@gasket\/plugin-)/;
 let _metadata;
 
-function tryRequire(path) {
+function tryRequire(tryPath) {
   try {
-    return require(path);
+    return require(tryPath);
```

Comment:
The main reason this plugin should be is the first ESM port candidate is to handle loading metadata from other ESM plugins. Having a `tryRequire` for the `package.json` files _should_ be fine though. Is that the only use case `require()` besides `require.resolve()`?

Reply:
Yes, require is using only for package.json and resolve. I saw this pattern in other plugins to handle package.json imports.

Reply:
Yeah it's to get around the import attributes. If this works I'm good with it 👍 . I've never tried `require.resolve` with `createRequire` before but I imagine it would work.

Reply:
Cool - I linked it into a working app and confirmed it works with the `docs` command:

<img width="546" alt="Screenshot 2024-11-21 at 6 08 14 PM" src="https://github.com/user-attachments/assets/e53dd3a7-6fc4-469a-8921-fb0a2300411d">